### PR TITLE
JAMES-2868 MockSmtpServer http route for getting all mails

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

File: server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/HTTPConfigurationServer.java [hunk 1]
```diff
@@ -77,23 +79,49 @@ protected void doDelete(HttpServletRequest req, HttpServletResponse resp) {
         }
     }
 
+    static class SMTPMailsServlet extends HttpServlet {
+        private final ObjectMapper objectMapper;
+        private final ReceivedMailRepository receivedMailRepository;
+
+        SMTPMailsServlet(ReceivedMailRepository receivedMailRepository) {
+            this.objectMapper = new ObjectMapper()
```
Comment (chibenwa) [inline server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/HTTPConfigurationServer.java]: Why do we duplicate the object mapper in each servlet?

File: server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/jackson/MailAddressModule.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+package org.apache.james.mock.smtp.server.jackson;
+
+import org.apache.james.core.MailAddress;
+import org.apache.james.webadmin.utils.JsonTransformerModule;
+
+import com.fasterxml.jackson.databind.Module;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+
+public class MailAddressModule implements JsonTransformerModule {
```
Comment (chibenwa) [inline server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/jackson/MailAddressModule.java]: Please do an interface with a constant for the module and avoid relying on webadmin dependency (hunt that dependency down and remove it from the POM!)

File: server/mailet/mock-smtp-server/pom.xml [hunk 3]
```diff
@@ -54,6 +54,10 @@
             <artifactId>james-server-testing</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-webadmin-core</artifactId>
```
Comment (chibenwa) [inline server/mailet/mock-smtp-server/pom.xml]: Let's remove that!

File: server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/jackson/MailAddressModule.java [hunk 4]
```diff
@@ -20,12 +20,11 @@
 package org.apache.james.mock.smtp.server.jackson;
 
 import org.apache.james.core.MailAddress;
-import org.apache.james.webadmin.utils.JsonTransformerModule;
 
 import com.fasterxml.jackson.databind.Module;
 import com.fasterxml.jackson.databind.module.SimpleModule;
 
-public class MailAddressModule implements JsonTransformerModule {
+public class MailAddressModule implements JsonSimpleModule {
```
Comment (chibenwa) [inline server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/jackson/MailAddressModule.java]:
```
public interface MailAddressJacksonModule {
    SimpleModule MODULE = new SimpleModule()
            .addSerializer(MailAddress.class, new MailAddressSerializer())
            .addDeserializer(MailAddress.class, new MailAddressDeserializer());
}
```

then:

```
   private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper()
        .registerModule(new Jdk8Module())
        .registerModule(new GuavaModule())
        .registerModule(MailAddressJacksonModule.MODULE);
```

And we can remove the `JsonSimpleModule` interface...

**Comment by chibenwa** [comment]
```
[a8fb88ead842893e8b94113f0253d754c2951f52] ogres[ERROR] /james-project/server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/Fixture.java:[44,25] org.apache.james.mock.smtp.server.jackson.MailAddressModule is abstract; cannot be instantiated
[a8fb88ead842893e8b94113f0253d754c2951f52] [ERROR] /james-project/server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/jackson/MailAddressModuleTest.java:[39,29] org.apache.james.mock.smtp.server.jackson.MailAddressModule is abstract; cannot be instantiated
```

File: server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/HTTPConfigurationServerTest.java [hunk 5]
```diff
@@ -44,87 +54,178 @@
...
-            .body(JSON_BEHAVIORS)
-        .when()
-            .put()
-        .then()
-            .statusCode(HttpStatus.NO_CONTENT_204);
-    }
+            String response = when()
+                .get()
+                .then()
+                .extract().asString();
```
Comment (blackheaven) [inline server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/HTTPConfigurationServerTest.java]: CR

File: server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/HTTPConfigurationServerTest.java [hunk 6]
```diff
@@ -44,87 +54,178 @@
...
-    }
+    @Nested
+    class SMTPBehaviorsTest {
+        @BeforeEach
+        void setUp() throws Exception {
+            server = HTTPConfigurationServer.onRandomPort(new SMTPBehaviorRepository(), new ReceivedMailRepository());
+            server.start();
+
+            RestAssured.requestSpecification = new RequestSpecBuilder()
+                .setContentType(ContentType.JSON)
```
Comment (mbaechler) [inline server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/HTTPConfigurationServerTest.java]: I don't think you assert the incoming content type

**Comment by chibenwa** [comment]
merged

**Pull request closed without merging**